lib/streamlit/web/server/browser_websocket_handler.py

```diff
@@ -104,6 +104,14 @@ def open(self, *args, **kwargs) -> Optional[Awaitable[None]]:
         else:
             user_info["email"] = email
 
+        remote_ip = (
+            self.request.headers.get("X-Real-IP")
+            or self.request.headers.get("X-Forwarded-For")
```

Comment:
This is not safe, because the X-Forwarded-For and X-RealIP headers can be set by the end user.

Reply:
How about add an option to allow user use header instead of ip? If user deploy streamlit behind a reverse proxy than use header will be safe.

Reply:
It certainly has nothing to do with security. It is just information to be accessed by anyone who wants it.